doc/dualtor/active_active_hld.md

```diff
@@ -0,0 +1,448 @@
+# Active-Active Dual ToR
+
+Active-active dual ToR link manager is an evolution of active-standby dual ToR link manager. Both ToRs are expected to handle traffic in normal scenarios. For consistency, we will keep using the term "standby" to refer inactive links or ToRs. 
+
+## Revision
+
+| Rev | Date     | Author          | Change Description |
+|:---:|:--------:|:---------------:|--------------------|
+| 0.1 | 05/23/22 | Jing Zhang      | Initial version    |
+
+## Scope 
+This document provides the high level design of SONiC dual toR solution, supporting active-active setup. 
+
+## Content
+[1 Cluster Topology](#active_active_hld.md#1-cluster-topology)  
+
+[2 Requrement Overview](#2-requrement-overview)  
+  - [2.1 Server Requirements](#21-server-requirements)
+  - [2.2 SONiC Requirements](#22-sonic-requirements)  
+
+[3 SONiC ToR Controlled Solution](#3-sonic-tor-controlled-solution)  
+  - [3.1 IP Routing](#31-ip-routing)
+    - [3.1.1 Normal Scenario](#311-normal-scenario)
+    - [3.1.2 Server Uplink Issue](#312-server-uplink-issue)
+    - [3.1.3 ToR Failure](#313-tor-failure)
+    - [3.1.4 Comparison to Active-Standby ](#314-comparison-to-active-standby)
+  - [3.2 DB Schema Changes](#32-db-schema-changes)
+    - [3.2.1 Config DB](#321-config-db)
+    - [3.2.2 App DB](#322-app-db)
+    - [3.2.3 State DB](#323-state-db)
+  - [3.3 Linkmgrd](#33-linkmgrd)
+    - [3.3.1 Link Prober](#331-link-prober)
+    - [3.3.2 Link State](#332-link-state)
+    - [3.3.3 Forwarding State](#333-forwarding-state)
+    - [3.3.4 Acitve-Active State Machine](#334-acitve-active-state-machine)
+    - [3.3.5 Default route to T1](#335-default-route-to-t1)
+    - [3.3.6 Incremental Featrues](#336-incremental-featrues)
+  - [3.4 Orchagent](#34-orchagent)
+    - [3.4.1 IPinIP tunnel](#341-ipinip-tunnel)
+    - [3.4.2 Flow Diagram and Orch Components](#342-flow-diagram-and-orch-components)
+  - [3.5 Transceiver Daemon](#35-transceiver-daemon)
+    - [3.5.1 Cable Control through gRPC](#351-cable-control-through-grpc)
+  - [3.6 State Transition Flow](#36-state-transition-flow)
+  - [3.7 Further Enhancement](#37-further-enhancement)
+  - [3.8 Command Line](#38-command-line)
+
+[4 Warm Reboot Support](#4-warm-reboot-support)
+
+## 1 Cluster Topology 
+There are a certain number of racks in a row, each rack will have 2 ToRs, and each row will have 8 Tier One (T1s) network devices. Each server will have a NIC connected to 2 ToRs with 100 Gbps DAC cables. 
+
+In this design:
+* Both upper ToR (labeled as UT0) and lower ToR (labeled as LT0) will advertise same VLAN to upstream T1s, each T1 will see 2 available next hops for the VLAN. 
```

Comment:
what we meant by "advertise same VLAN"?  Did you mean IP address instead?

Reply:
Yes, thanks for pointing it out. Updated accordingly.